### DISCUSSION TITLE
local edge function doesn't seem to work.

### DISCUSSION THREAD

**Discussion by jeanengelbrecht02** [supabase/supabase#22690] [Questions]
In the docs it says that if you write console.log() in your edge function and then invoke it ,  the logs should be in the terminal. 

this doesn't seem to be the case for me.   supabase start also seems to return,  it does give the anon ID but the command still finishes.  So it doesnt actually keep open a way for the logs to return in the first place.  

So basically where and how do i read logs from my local edge function using supabase start.

**Comment by alexgshaw** [comment]
Running into the same issue.

**Comment by glennslaven** [comment]
You need to run `supabase functions serve` after running `supabase start` to start the local hosting of the edge-functions. The console output will be displayed in that terminal.

**Comment by alexgshaw** [comment]
Let me add some detail on the problem I'm facing.

I start supabase using `supabase start`

I then start my supabase edge functions using `supabase functions serve --env-file supabase/.env.local`

Everything up until now works fine.

Originally, I noticed that when I added a `console.log` statement to my file, it was not printing to the terminal as I expected.

I then noticed that no changes were being reflected in my locally running function. I then completely deleted the function. It still ran as before, even though it doesn't even exist anymore.

So then, I stopped my local edge function server and started it again with `supabase functions serve --env-file supabase/.env.local`. It still did not reflect the fact that I deleted the function.

Even stopping the server, running `supabase stop`, and then starting and serving the functions again did not work. To finally get the function changes to reflect, I had to restart my computer. But now when I make additional changes, they still don't reflect. This must be a bug. I'm not sure where the functions are cached, but the cache is not updating.

Supabase CLI: 1.169.8
Deno: 1.40.3

Any pointers?

**Comment by rizhangdev** [comment]
Running into this same issue... I started a new supabase edge function project, which never encountered this issue.
